src/core/components/object.js

```diff
@@ -333,7 +333,7 @@ module.exports = function object (self) {
         if (cid.codec === 'dag-cbor') {
           const links = findLinks(result)
 
-          callback(null, links)
+          return callback(null, links)
```

Comment:
Good catch I've opened a separate PR for this here https://github.com/ipfs/js-ipfs/pull/2111